.github/workflows/test-wdio.yml

```diff
@@ -0,0 +1,45 @@
+name: WebdriverIO Tests
+
+on:
+  merge_group:
+  pull_request:
+  push:
+    branches:
+      - 'main'
+
+jobs:
+  build_core:
+    name: Build
+    uses: ./.github/workflows/build.yml
+
+  run_browserstack:
+    name: Run WebdriverIO Component Tests
+    runs-on: ubuntu-22.04
+    needs: [build_core]
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Use Node Version from Volta (Karma)
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
+        with:
+          # pull the version to use from the volta key in package.json
+          node-version-file: './test/karma/package.json'
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Download Build Archive
+        uses: ./.github/workflows/actions/download-archive
+        with:
+          name: stencil-core
+          path: .
+          filename: stencil-core-build.zip
+
+      - name: Run WebdriverIO Component Tests
+        run: npm run test.wdio
```

Comment:
I think it's sort of six of one half a dozen of another but would it make sense for us to kick off the wdio tests from within the `main` workflow, like we do for the unit tests and whatnot? instead of having a separate workflow here that runs independently

Reply:
also is the plan to circle back later to do some sort of matrix on browsers and browser versions?

Reply:
> to kick off the wdio tests from within the main workflow, like we do for the unit tests and whatnot?

The test run is cheap and can be run as part of the unit test.

> also is the plan to circle back later to do some sort of matrix on browsers and browser versions?

Just pushed a fix to include Firefox and Safari when running in CI

Reply:
I recognize that there are some browser differences with some assertions. I had to change some assertion values based on whether we run in Safari or not. This could be either an actual browser issue or something with the driver underneath. I would like to investigate this separately. This could become ultimately a burden to maintain at some point. I can bring this up at the next Jam session.